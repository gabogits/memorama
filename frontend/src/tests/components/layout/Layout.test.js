import { mount } from "enzyme";
import { Router } from "react-router-dom";
import Layout from "../../../components/layout/Layout";
import GameContext from "../../../context/GameContext";

describe("Pruebas en <Layout />", () => {
  let historyMock = {
    push: jest.fn(),
    replace: jest.fn(),
    location: jest.fn(),
    listen: jest.fn(),
    createHref: jest.fn(),
  };

  afterEach(() => {
    jest.clearAllMocks();
  });

  const contextValue = {
    dispatch: jest.fn(),
    state: {
      numberPlayers: 2,
      dataPlayers: [
        { avatar: "witch", id: 1, name: "juan" },
        { avatar: "dracula", id: 2, name: "bety" },
      ],
      arrayImages: [],
      category: null,
    },
  };

  test("Debe de mostarrse correctamente el component en el layout", () => {
    const Children = <div className="contenido">Hola soy contenido</div>;
    const wrapper = mount(
      <GameContext.Provider value={contextValue}>
        <Router history={historyMock}>
          <Layout children={Children} />
        </Router>
      </GameContext.Provider>
    );
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find(".contenido").text().trim()).toBe("Hola soy contenido");
  });
});
