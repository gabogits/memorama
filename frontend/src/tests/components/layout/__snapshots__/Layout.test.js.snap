// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <Layout /> Debe de mostarrse correctamente el component en el layout 1`] = `
<main>
  <header>
    <div
      className=" container container-header"
    >
      <div
        className="logo"
      >
        <figure>
          <a
            onClick={[Function]}
          >
            <span
              className="logo-desk"
            >
              MEMORAMA
            </span>
            <span
              className="logo-mobile"
            >
              M
            </span>
             
          </a>
        </figure>
      </div>
      <nav>
        <div
          className="nav-left"
        />
        <div
          className="nav-right"
        >
          <div
            className="nav-list"
          >
            <ul>
              <li>
                <button
                  onClick={[Function]}
                >
                  Inicio
                </button>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Tema
                </a>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Juego
                </a>
              </li>
              <li>
                <a
                  aria-current={null}
                  onClick={[Function]}
                >
                  Ranking
                </a>
              </li>
            </ul>
          </div>
        </div>
      </nav>
    </div>
  </header>
  <div
    className="contenido"
  >
    Hola soy contenido
  </div>
</main>
`;
